php/libraries/Utility.class.inc

```diff
@@ -75,9 +84,10 @@ class Utility
      */
     static function crunch($x, $y)
     {
-        $a = (is_array($x)) ? array_reduce($x, array('Utility', 'crunch')) : $x;
-        $b = (is_array($y)) ? array_reduce($y, array('Utility', 'crunch')) : $y;
-        return (!empty($a)) ? "$a`$b" : $b;
+        throw new LorisException(
+            "Utility::crunch has been deprecated. "
+            . "If really needed, consider implementing your own."
+        );
```

Comment:
There's a more specific `DeprecatedException` that's intended to be used in cases like this. There should also be a `@deprecated` doc comment tag, so that phan can find invalid calls to it.

I'd also remove the second line. It doesn't give the user any useful information.

(same comments apply to `reduce`)